xdsl/transforms/experimental/stencil_global_to_local.py

```diff
@@ -0,0 +1,72 @@
+from dataclasses import dataclass
+from typing import TypeVar
+from abc import ABC, abstractmethod
+
+from xdsl.pattern_rewriter import (PatternRewriter, PatternRewriteWalker,
+                                   RewritePattern, GreedyRewritePatternApplier,
+                                   op_type_rewrite_pattern)
+from xdsl.ir import MLContext
+from xdsl.irdl import Attribute
+from xdsl.dialects import builtin
+from xdsl.dialects.experimental import stencil
+
+_T = TypeVar('_T', bound=Attribute)
+
+
+@dataclass
+class SlicingStrategy(ABC):
```

Comment:
Nice idea of slicing strategy, in terms of terminology maybe instead "DecompositionStrategy" (if you decompose in 2D or 3D you aren't really slicing but instead cubing etc...)